python_modules/libraries/dagster-fivetran/dagster_fivetran/resources.py

```diff
@@ -930,10 +941,107 @@ def load_asset_specs(
             workspace=self, dagster_fivetran_translator=dagster_fivetran_translator.__class__
         )
 
-    def sync_and_poll(
-        self, context: Optional[Union[OpExecutionContext, AssetExecutionContext]] = None
+    def _generate_materialization(
+        self,
+        fivetran_output: FivetranOutput,
+        dagster_fivetran_translator: DagsterFivetranTranslator,
     ):
-        raise NotImplementedError()
+        connector = FivetranConnector.from_connector_details(
+            connector_details=fivetran_output.connector_details
+        )
+        schema_config = FivetranSchemaConfig.from_schema_config_details(
+            schema_config_details=fivetran_output.schema_config
+        )
+
+        for schema_source_name, schema in schema_config.schemas.items():
+            if not schema.enabled:
+                continue
+
+            for table_source_name, table in schema.tables.items():
+                if not table.enabled:
+                    continue
+
+                asset_key = dagster_fivetran_translator.get_asset_spec(
+                    props=FivetranConnectorTableProps(
```

Comment:
We reconstruct a `FivetranConnectorTableProps` to get the asset key using `get_asset_spec(...).key`